modules/ui/maproulette_details.js

```diff
@@ -1,79 +1,131 @@
     const detailsEnter = details.enter()
       .append('div')
       .attr('class', 'error-details qa-details-container');
 
-    detailsEnter
-      .append('div')
+    detailsEnter.append('div')
       .attr('class', 'qa-details-subsection')
       .text(l10n.t('map_data.layers.maproulette.loading_task_details'));
 
-    // update
     details = details.merge(detailsEnter);
 
-
-    maproulette.loadTaskDetailAsync(_qaItem)
-      .then(task => {
-        // Do nothing if _qaItem has changed by the time Promise resolves
-        if (_qaItem.id !== task.id) return;
-
-        const selection = details.selectAll('.qa-details-subsection');
-        selection.html('');   // replace contents
-
-        // Things like keys and values are dynamically added to a subtitle string
-        if (task.id) {
-          selection
-            .append('h4')
-            .text(l10n.t('map_data.layers.maproulette.id_title'));
-
-          selection
-            .append('p')
-            .text(`${task.parentId} / ${task.id}`)
-            .selectAll('a')
-            .attr('rel', 'noopener')
-            .attr('target', '_blank');
-        }
-
-        if (task.description && !task.description.includes('Lorem')) {
-          selection
-            .append('h4')
-            .text(l10n.t('map_data.layers.maproulette.detail_title'));
-
-          selection
-            .append('p')
-            .html(task.description)  // parsed markdown
-            .selectAll('a')
-            .attr('rel', 'noopener')
-            .attr('target', '_blank');
-        }
-
-        if (task.instruction && !task.instruction.includes('Lorem') && task.instruction !== task.description) {
-          selection
-            .append('h4')
-            .text(l10n.t('map_data.layers.maproulette.instruction_title'));
-
-          selection
-            .append('p')
-            .html(task.instruction)  // parsed markdown
-            .selectAll('a')
-            .attr('rel', 'noopener')
-            .attr('target', '_blank');
-        }
-      })
-      .catch(e => console.error(e));  // eslint-disable-line
+    maproulette.loadTaskDetailAsync(_qaItem).then(task => {
+      if (!task) return;
+      if (_qaItem.id !== task.id) return;
+      const selection = details.selectAll('.qa-details-subsection');
+      selection.html('');   // replace contents
+      // Display Challenge ID and Task ID
+      if (task.id) {
+        selection
+          .append('h4')
+          .text(l10n.t('map_data.layers.maproulette.id_title'));
+        selection
+          .append('p')
+          .text(`${task.parentId} / ${task.id}`)
+          .selectAll('a')
+          .attr('rel', 'noopener')
+          .attr('target', '_blank');
+      }
+
+      const description = parseShortCodes(replaceMustacheTags(task.description, task));
+      const instruction = parseShortCodes(replaceMustacheTags(task.instruction, task));
+      if (task.description && !task.description.includes('Lorem')) {
```

Comment:
Not sure about the 'lorem' check here- is this a default string for a new, pristine task?